src/components/Avatar.js

```diff
@@ -30,6 +30,14 @@ class Avatar extends Component {
     }
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.username) {
```

Comment:
Check `this.props.username !== nextProps.username` to avoid unnecesary setStates?

Reply:
if `nextprops.username` is falsy, should the defaultImage been shown?

Reply:
I added the `this.props.username !== nextProps.username` check.
We shouldn't set the default image state here, even if `nextProps.username` is falsy. We want it to hit the `img.busy.org` and it will get the default avatar on there.. the `defaultImage` is only used when `img.busy.org` fails

Reply:
Seems reasonable.

Then in `componentWillMount` is necessary the check? in terms of consistency seems strange that one checks it and other doesn't.

Reply:
nope, just removed it